pkg/controller/usersignup/usersignup_controller.go

```diff
@@ -192,13 +203,13 @@ func (r *ReconcileUserSignup) Reconcile(request reconcile.Request) (reconcile.Re
 	// status is set to false. This is especially important for cases when a user is deactivated and then reactivated
 	// because the status is used to trigger sending of the notification. If a user is reactivated a notification should
 	// be sent to the user again.
-	if !banned && !states.Deactivating(userSignup) && !userSignup.Spec.Deactivated {
+	if !banned && !states.Deactivating(userSignup) && !states.Deactivated(userSignup) {
 		if err := r.updateStatus(logger, userSignup, r.setStatusDeactivatingNotificationNotInPreDeactivation); err != nil {
 			return reconcile.Result{}, err
 		}
 	}
 
-	if states.Deactivating(userSignup) && !userSignup.Spec.Deactivated && condition.IsNotTrue(userSignup.Status.Conditions,
+	if states.Deactivating(userSignup) && !states.Deactivated(userSignup) && condition.IsNotTrue(userSignup.Status.Conditions,
```

Comment:
Shouldn't we have only one "deactivating" or "deactivated" state? When setting "deactivating" we should delete "deactivated". And when setting "deactivated" we should delete "deactivating". WDYT?

Reply:
Yes that makes sense, I'll update it.

Reply:
That's [done now](https://github.com/codeready-toolchain/toolchain-common/pull/162/commits/948073578b628ae9587dd48fd4aa3165ba0f193d).  Once I get approval for the toolchain-common PR then I can proceed with addressing the other comments on this PR.

Reply:
I've also removed the condition check for Deactivated.